core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/io/SdkInputStream.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.internal.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.utils.IoUtils;
+
+/**
+ * Base class for AWS Java SDK specific {@link InputStream}.
+ */
+@SdkInternalApi
+public abstract class SdkInputStream extends InputStream implements Releasable {
```

Comment:
Do we need this? I don't see anything using it, but I could be missing something.

Reply:
The `ChunkedEncodedInputStream` extends it.

Reply:
I see that, but is it relying on any specific implementation here? The methods don't see to be called, for example.

Reply:
Yeah, fair. The existing implementation extends `SdkInputStream`, so kept it that way. I may end up calling `abortIfNeeded` in some places. What are your thoughts on that? Do we really need it?

Reply:
I'd leave it out until we need it.